src/java/org/apache/cassandra/index/SecondaryIndexManager.java

```diff
@@ -1036,41 +1071,50 @@ public Index getBestIndexFor(RowFilter rowFilter)
                 RowFilter.CustomExpression customExpression = (RowFilter.CustomExpression) expression;
                 logger.trace("Command contains a custom index expression, using target index {}", customExpression.getTargetIndex().name);
                 Tracing.trace("Command contains a custom index expression, using target index {}", customExpression.getTargetIndex().name);
-                return indexes.get(customExpression.getTargetIndex().name);
-            }
-            else if (!expression.isUserDefined())
-            {
-                indexes.values().stream()
-                       .filter(index -> index.supportsExpression(expression.column(), expression.operator()))
-                       .forEach(searchableIndexes::add);
+                 indexes.get(customExpression.getTargetIndex().name);
```

Comment:
nit: extra space

Reply:
Actually, do we need this at all? I think this was leftover from the previous implementation.

Reply:
you are right.. it should be removed.